### PR TITLE
[receiver/sqlquery] fail on missing value for logs, collect errors

### LINKED ISSUE #35068: [receiver/sqlquery] Improve error handling for missing columns in result set
### Component(s)

internal/sqlquery, receiver/sqlquery

### Describe the issue you're reporting

I have two suggestions how to improve error handling:

1. The value and body columns do not match the column names in the result set, handling is different for logs and metrics. Here is the problematic configuration:
```yaml
  sqlquery:
...
      - sql: "select max(datapoint) as datapoint from logs_data group by log_type"
        metrics:
          - metric_name: logs.datapoint
            value_column: "datapoint_ne"
      - sql: "select 'Max datapoint value for ' || log_type || ' is ' || max(datapoint) as body from logs_data group by log_type"
        logs:
          - body_column: body_ne
```
When I run the Otel collector with this configuration, logs are generated (with an empty body), but metrics are not. Below is the collector's output:
```json
{
  "kind": "receiver",
  "name": "sqlquery",
  "data_type": "metrics",
  "error": "row 0: rowToMetric: value_column 'datapoint_ne' not found in result set\nrow 1: rowToMetric: value_column 'datapoint_ne' not found in result set\nrow 2: rowToMetric: value_column 'datapoint_ne' not found in result set",
  "scraper": "sqlqueryreceiver/query-0: select max(datapoint) as datapoint from logs_data group by log_type"
}
```
Notably, there is no error entry for "data_type": "logs". Ideally, the handling of missing body columns for logs should be consistent with the handling of missing value columns for metrics. If we choose to fail on a missing body column, it would be a breaking change.

2. The attribute_columns do not match the column names in the result set, error is logged only for the first miss. Here is the faulty configuration::
```yaml
  sqlquery:
...
    queries:
      - sql: "select 'Max datapoint value for ' || log_type || ' is ' || max(datapoint) as datapoint, log_type from logs_data group by log_type"
        logs:
          - body_column: datapoint
            attribute_columns: [ "log_type_ne_1", "log_type_ne_2" ]
```
When the collector starts, the following error is logged:
```json
{
  "kind": "receiver",
  "name": "sqlquery",
  "data_type": "logs",
  "error": "rowToLog: attribute_column not found: 'log_type_ne_1'\nrowToLog: attribute_column not found: 'log_type_ne_1'\nrowToLog: attribute_column not found: 'log_type_ne_1'",
  "query": "query-0: select 'Max datapoint value for ' || log_type || ' is ' || max(datapoint) as datapoint, log_type from logs_data group by log_type"
}
```
Although both attribute columns are missing from the result set, the error is logged only for the first missing attribute column. Providing more detailed error messages would enhance the debugging experience.

### PR DESCRIPTION
**Pull request by Grandys**
**Description:** 

* [breaking] Fail if for log column not found in result set (for consistency with metrics behaviour)
* Instead of fail-fast, collect all errors that occurred when transforming row to metric or log

**Link to tracking Issue:** #35068

**Testing:** Added/updated unit tests

**Documentation:** n/a

Closes #35068

### CODE REVIEW

File: receiver/sqlqueryreceiver/logs_receiver.go [hunk 1]
```diff
@@ -316,16 +316,24 @@ func (queryReceiver *logsQueryReceiver) storeTrackingValue(ctx context.Context,
 }
 
 func rowToLog(row sqlquery.StringMap, config sqlquery.LogsCfg, logRecord plog.LogRecord) error {
-	logRecord.Body().SetStr(row[config.BodyColumn])
+	var errs []error
+	var bodyColumnName = config.BodyColumn
```
Comment (crobert-1) [inline receiver/sqlqueryreceiver/logs_receiver.go:320]: I don't see any extra benefit to using a temporary variable here so I'd prefer to just use `config.BodyColumn`, unless I'm missing something.

**Review by crobert-1 — APPROVED** [review]

File: receiver/sqlqueryreceiver/logs_receiver.go (hunk 1 above)
Comment (Grandys) [reply receiver/sqlqueryreceiver/logs_receiver.go:320]: Temporary variable is also used in line [323](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35189/files#diff-0c9a67fda52566bc14742cf772b15a23ac8c74c66e5fc4f9b86c734b809d4a09R323), in the error.

Please confirm if you prefer two calls to `config.BodyColumn`, or if I should leave temporary variable.

Comment (crobert-1) [reply receiver/sqlqueryreceiver/logs_receiver.go:320]: Yes, I'd prefer to just use `config.BodyColumn` and remove the temporary variable 👍

Comment (Grandys) [reply receiver/sqlqueryreceiver/logs_receiver.go]: Temporary variable removed in [69e8f54](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35189/commits/69e8f54276f79d3081796de6c3269d4d73f6d8e6)

**Review by crobert-1 — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]
Love it! Thank you!

**Pull request merged**